### DISCUSSION TITLE
Wildcard SSL Certificate for local container and devices

### DISCUSSION THREAD

**Discussion by techsolo12** [NginxProxyManager/nginx-proxy-manager#2358] [Q&A]
Hello guys,

today i wanted to setup some SSL certificates for my homelab.

Some informations to my the software in my homelab:
I use a opensense as firewall with installed outbound as dns resolver.
In opnsense i have configured my domain as local domain, something like "meinedomain.de".
After that i set for pihole and some other software host overrides with there hostnames and ips.
On all devices i set pihole as DNS. In pihole i set opnsense as DNS.
Everything works fine. I can browse piholes ui with "pihole.meinedomain.de/admin"

Now i want to setup up a SSL certificate for pihole with my nginx proxy manager.
So i opened the webui and setup a wildcard certificate with dns challenge on cloudflare.
I got a certificate which works perfectly. My cloud and other webservices are protected.

But when i setup up a proxy host for pihole its not working. :/ I cant get a secure connection to pihole.
On the pictures you can see my setup. I hope that anybody of you can help me to troubleshoot the problem. 
A possible solution where to install on every host the acme.sh script to got a certificate, but this will be a lot of work. 
The perfect solution is to use nginx proxy manager for the ssl management.

Im sorry for my bad english :/ But hopefully you can help me :)

![Unbenannt](https://user-images.githubusercontent.com/36322543/199596659-37566c82-72bc-4bc0-a4c2-fdeddec2e091.JPG)
![Unbenannt1](https://user-images.githubusercontent.com/36322543/199596683-913485cb-d0f1-4ca9-8c8a-f6cc0a2e4841.JPG)
![Unbenannt2](https://user-images.githubusercontent.com/36322543/199596703-b15ac423-774e-4a24-970b-52d9b127b83c.JPG)

**Comment by G0lfH0t3l** [comment]
Just a small remark from another NPM/Pi-hole user:
It works for me with a wildcard certificate, i've added this in the advanced box of the proxy host:
location = /{return 301 $scheme://$http_host/admin/;}

In my setup I did not add a location under Custom locations.
